x-pack/plugins/fleet/server/services/epm/archive/extract.ts

```diff
@@ -19,14 +21,22 @@ export async function untarBuffer(
 ): Promise<unknown> {
   const deflatedStream = bufferToStream(buffer);
   // use tar.list vs .extract to avoid writing to disk
-  const inflateStream = tar.list().on('entry', (entry: tar.FileStat) => {
-    const path = entry.header.path || '';
+  const inflateStream = tar.list().on('entry', (entry) => {
+    const path = entry.path || '';
     if (!filter({ path })) return;
-    streamToBuffer(entry).then((entryBuffer) => onEntry({ buffer: entryBuffer, path }));
+    streamToBuffer(entry as unknown as NodeJS.ReadableStream).then((entryBuffer) =>
```

Comment:
`streamToBuffer` expects a `ReadableStream`, but `entry` isn't one. The types seem to be a little to strict though and casting like this should appease TS.